src/WallToolPaths.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2020 Ultimaker B.V.
+// CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#include "WallToolPaths.h"
+
+#include "SkeletalTrapezoidation.h"
+#include "utils/polygonUtils.h"
+
+namespace cura
+{
+constexpr coord_t transition_length_multiplier = 2;
+
+WallToolPaths::WallToolPaths(const Polygons& outline, const coord_t nominal_bead_width, const coord_t inset_count,
+                             const Settings& settings)
+    : outline(outline)
+    , nominal_bead_width(nominal_bead_width)
+    , inset_count(inset_count)
+    , strategy_type(settings.get<StrategyType>("beading_strategy_type"))
+    , widening_beading_enabled(settings.get<bool>("widening_beading_enabled"))
+    , min_bead_width(widening_beading_enabled ? new coord_t(settings.get<coord_t>("min_bead_width")) : nullptr)
+    , min_feature_size(widening_beading_enabled ? new coord_t(settings.get<coord_t>("min_feature_size")) : nullptr)
+    , small_area_length(INT2MM(static_cast<double>(nominal_bead_width) / 2))
+    , transition_length(transition_length_multiplier * nominal_bead_width)
+    , toolpaths_generated(false)
+{
+}
+
+const ToolPaths& WallToolPaths::generate()
+{
+    constexpr coord_t smallest_segment = 50;
+    constexpr coord_t allowed_distance = 50;
+    constexpr coord_t epsilon_offset = (allowed_distance / 2) - 1;
+    constexpr float transitioning_angle = 0.5;
+
+    Polygons prepared_outline = outline.offset(-epsilon_offset).offset(epsilon_offset);
+    prepared_outline.simplify(smallest_segment, allowed_distance);
+    PolygonUtils::fixSelfIntersections(epsilon_offset, prepared_outline);
+    prepared_outline.removeDegenerateVerts();
+    prepared_outline.removeColinearEdges();
+    // TODO: complete guess as to when arachne starts breaking, but it doesn't  function well when an area is really
+    // small apparently?
+    prepared_outline.removeSmallAreas(small_area_length * small_area_length, false);
```

Comment:
Was this code copied from somewhere?

Reply:
@rburema his implementation

Reply:
Ah I see it was moved away from `WallsComputation` to this new WallToolPaths. Fine then, at least no code duplication.